src/ImageSharp/Formats/WebP/AlphaDecoder.cs

```diff
@@ -0,0 +1,309 @@
+        public int Width { get; }
+
+        /// <summary>
+        /// Gets the height of the image.
+        /// </summary>
+        public int Height { get; }
+
+        /// <summary>
+        /// Gets the used filter type.
+        /// </summary>
+        public WebPAlphaFilterType AlphaFilterType { get; }
+
+        /// <summary>
+        /// Gets or sets the last decoded row.
+        /// </summary>
+        public int LastRow { get; set; }
+
+        /// <summary>
+        /// Gets or sets the row before the last decoded row.
+        /// </summary>
+        public int PrevRow { get; set; }
+
+        /// <summary>
+        /// Gets information for decoding Vp8L compressed alpha data.
+        /// </summary>
+        public Vp8LDecoder Vp8LDec { get; }
+
+        /// <summary>
+        /// Gets the decoded alpha data.
+        /// </summary>
+        public IMemoryOwner<byte> Alpha { get; }
+
+        public int CropTop { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether pre-processing was used during compression.
+        /// 0: no pre-processing, 1: level reduction.
+        /// </summary>
+        private int PreProcessing { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the alpha channel uses compression.
+        /// </summary>
+        private bool Compressed { get; }
+
+        /// <summary>
+        /// Gets the (maybe compressed) alpha data.
+        /// </summary>
+        private byte[] Data { get; }
+
+        /// <summary>
+        /// Gets the Vp8L decoder which is used to de compress the alpha channel, if needed.
+        /// </summary>
+        private WebPLosslessDecoder LosslessDecoder { get; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the decoding needs 1 byte per pixel for decoding.
+        /// Although Alpha Channel requires only 1 byte per pixel, sometimes Vp8LDecoder may need to allocate
+        /// 4 bytes per pixel internally during decode.
+        /// </summary>
+        public bool Use8BDecode { get; set; }
+
+        /// <summary>
+        /// Decodes and filters the maybe compressed alpha data.
+        /// </summary>
+        public void Decode()
+        {
+            if (this.Compressed is false)
+            {
+                if (this.Data.Length < (this.Width * this.Height))
+                {
+                    WebPThrowHelper.ThrowImageFormatException("not enough data in the ALPH chunk");
+                }
+
+                Span<byte> alphaSpan = this.Alpha.Memory.Span;
+                if (this.AlphaFilterType == WebPAlphaFilterType.None)
+                {
+                    this.Data.AsSpan(0, this.Width * this.Height).CopyTo(alphaSpan);
+                    return;
+                }
+
+                Span<byte> deltas = this.Data.AsSpan();
+                Span<byte> dst = alphaSpan;
+                Span<byte> prev = null;
+                for (int y = 0; y < this.Height; ++y)
```

Comment:
that would lead to duplicate code as the for loop would be in each switch statement. What i could do is use the same approach as in the reference code and use a delegate array for the unfilter methods and the filter type would be the index into that array.

The original code looks like this:

```
for (y = 0; y < num_rows; ++y) {
        WebPUnfilters[alph_dec->filter_](prev_line, deltas, dst, width);
        prev_line = dst;
        dst += width;
        deltas += width;
      }
```

Im not sure though, if i should do that as i think it hurts readability a bit.